domain/application/state/application_endpoint.go

```diff
@@ -18,9 +17,33 @@ import (
 	"github.com/juju/juju/core/network"
 	corerelation "github.com/juju/juju/core/relation"
 	applicationerrors "github.com/juju/juju/domain/application/errors"
-	internalerrors "github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/errors"
 )
 
+// GetApplicationEndpointBindings returns the mapping for each endpoint name and
+// the space ID it is bound to (or empty if unspecified). When no bindings are
+// stored for the application, defaults are returned.
+//
+// If no application is found, an error satisfying
+// [applicationerrors.ApplicationNotFound] is returned.
+func (st *State) GetApplicationEndpointBindings(ctx context.Context, appUUID coreapplication.ID) (map[string]string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	var result map[string]string
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var err error
+		result, err = st.getEndpointBindings(ctx, tx, appUUID)
+		return errors.Capture(err)
```

Comment:
you don't need the capture here. The outer scope provides the ref and saves a few more ms in the transaction.